src/utils.ts

```diff
@@ -0,0 +1,189 @@
+export function overwriteBytes(a: Uint8Array, b: Uint8Array, i = 0): void {
+  if (a.length < b.length + i) {
+    throw Error(
+      `Cannot copy bytes because the base byte array length is lesser (${a.length}) than the others (${b.length})`,
+    )
+  }
+
+  for (let index = 0; index < b.length; index++) {
+    a[index + i] = b[index]
+  }
+}
+
+/**
+ * Flattens the given array that consist of Uint8Arrays.
+ */
+export function flattenBytesArray(bytesArray: Uint8Array[]): Uint8Array {
+  if (bytesArray.length === 0) return new Uint8Array(0)
+
+  const bytesLength = bytesArray.map(v => v.length).reduce((sum, v) => (sum += v))
+  const flattenBytes = new Uint8Array(bytesLength)
+  let nextWriteIndex = 0
+  for (const b of bytesArray) {
+    overwriteBytes(flattenBytes, b, nextWriteIndex)
+    nextWriteIndex += b.length
+  }
+
+  return flattenBytes
+}
+
+export function equalBytes(a: Uint8Array, b: Uint8Array): boolean {
+  if (a.length !== b.length) return false
+
+  return a.every((byte, index) => b[index] === byte)
+}
+
+/**
+ * runs a XOR operation on data, encrypting it if it
+ * hasn't already been, and decrypting it if it has, using the key provided.
+ */
+export function encryptDecrypt(key: Uint8Array, data: Uint8Array, startIndex = 0, endIndex?: number): void {
+  endIndex ||= data.length
+
+  for (let i = startIndex; i < endIndex; i += key.length) {
+    const maxChunkIndex = i + key.length
+    const encryptionChunkEndIndex = maxChunkIndex <= data.length ? maxChunkIndex : data.length
+    const encryptionChunk = data.slice(i, encryptionChunkEndIndex)
+    for (let j = 0; j < encryptionChunk.length; j++) {
+      encryptionChunk[j] = Number(encryptionChunk[j]) ^ Number(key[j % key.length])
+    }
+    data.set(encryptionChunk, i)
+  }
+}
+
+export function keccak256Hash(...messages: Message[]): Bytes<32> {
+  const hasher = keccak256.create()
+
+  messages.forEach(bytes => hasher.update(bytes))
+
+  return Uint8Array.from(hasher.digest()) as Bytes<32>
+}
+
+/** Tested only for Uint16 BigEndian */
+export function fromBigEndian(bytes: Uint8Array): number {
+  if (bytes.length === 0) throw Error('fromBigEndian got 0 length bytes')
+  const numbers: number[] = []
+  const lastIndex = bytes.length - 1
+
+  for (let i = 0; i < bytes.length; i++) {
+    numbers.push(bytes[lastIndex - i] << (8 * i))
+  }
+
+  return numbers.reduce((bigEndian, num) => (bigEndian |= num))
+}
+
+/** Tested only with Uint16 BigEndian */
+export function toBigEndianFromUint16(value: number): Bytes<2> {
+  if (value < 0) throw Error(`toBigEndianFromUint16 got lesser than 0 value: ${value}`)
+  const maxValue = (1 << 16) - 1
+
+  if (value > maxValue) throw Error(`toBigEndianFromUint16 got greater value then ${maxValue}: ${value} `)
+
+  return new Uint8Array([value >> 8, value]) as Bytes<2>
+}
+
+export function gen32Bytes(): Bytes<32> {
+  return new TextEncoder().encode(nanoid(32)) as Bytes<32>
```

Comment:
Using `nanoid` this way is cryptographically less secure than the Bee version, because it truncates each byte to 6 bit when converting it to the URL safe alphabet. This is not a blocker issue now because there is no encryption implemented, but still it would be good to fix now, because it may be easy to overlook it when the encryption implementation will happen.

Using something like the [get-random-values](https://www.npmjs.com/package/get-random-values) package would solve this issue.